src/connectedk8s/azext_connectedk8s/custom.py

```
@@ -1381,3 +1157,347 @@ def _resolve_service_principal(client, identifier):  # Uses service principal gr
+                      client,
+                      resource_group_name,
+                      cluster_name,
+                      token=None,
+                      path=os.path.join(os.path.expanduser('~'), '.kube', 'config'),
+                      overwrite_existing=False,
+                      context_name=None,
+                      api_server_port=API_SERVER_PORT,
+                      client_proxy_port=CLIENT_PROXY_PORT):
+    
+    cloud=send_cloud_telemetry(cmd)
+    args=[]
+    operating_system=platform.system()
+
+    ##Creating installation location, request uri and older version exe location depending on OS
+    if(operating_system=='Windows') :
+        install_location_string=f'.clientproxy\\arcProxy{operating_system}{CLIENT_PROXY_VERSION}.exe'
+        requestUri=f'https://clientproxy.azureedge.net/release20201218/arcProxy{operating_system}{CLIENT_PROXY_VERSION}.exe'
+        older_version_string=f'.clientproxy\\arcProxy{operating_system}*.exe'
+        creds_string=r'.azure\accessTokens.json'
+        proc_name=f'arcProxy{operating_system}{CLIENT_PROXY_VERSION}.exe'
+    
+    elif(operating_system=='Linux' or operating_system=='Darwin') :
+        install_location_string=f'.clientproxy/arcProxy{operating_system}{CLIENT_PROXY_VERSION}'
+        requestUri=f'https://clientproxy.azureedge.net/release20201218/arcProxy{operating_system}{CLIENT_PROXY_VERSION}'
+        older_version_string=f'.clientproxy/arcProxy{operating_system}*'
+        creds_string=r'.azure/accessTokens.json'
+        proc_name=f'arcProxy{operating_system}{CLIENT_PROXY_VERSION}'
+
+    else :
+        telemetry.set_exception(exception='Unsupported OS', fault_type=consts.Unsupported_Fault_Type,
+                            summary=f'{operating_system} is not supported yet')
+        raise CLIError(f'The {operating_system} platform is not currently supported.')
+    
+    if(check_process(proc_name)) :
+        print('Another instance of proxy already running')
```

instead of print we should raise a cli error and add telemetry here as well

---

done